public/app/features/explore/state/query.ts

```diff
@@ -565,10 +565,12 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
     let newQuerySource: Observable<ExplorePanelData>;
     let newQuerySubscription: SubscriptionLike;
 
-    const queries = exploreItemState.queries.map((query) => ({
-      ...query,
-      datasource: query.datasource || datasourceInstance?.getRef(),
-    }));
+    const queries = exploreItemState.queries
```

Comment:
We need to "run" hidden queries (i.e. pass them the data source API) because they might be references to other queries (https://grafana.com/docs/grafana/latest/datasources/graphite/query-editor/#nest-queries). 

It's used by data sources to decide what should actually be run or not, e.g https://github.com/grafana/grafana/blob/6568f3669dce0c2e951a72bfa0833be1d1084838/public/app/plugins/datasource/loki/datasource.ts#L1115

It can be filtered in [here](https://github.com/grafana/grafana/blob/6568f3669dce0c2e951a72bfa0833be1d1084838/public/app/plugins/datasource/loki/datasource.ts#L266). It's annoying that each data source needs to take care or of it, but I think it's mainly to supported nested queries

Reply:
oh, wow

Reply:
thanks for giving me those directions